### PR TITLE
Modified region extraction with subregions and plink

### PR DESCRIPTION
**Pull request by tfabiha**
Region extraction implements extraction using plink files, however issue occurs with `plink_splicing`.

Region extraction as a whole implements extraction of subregions in order to minimize memory utilized at any given time.

### CODE REVIEW

**Comment by tfabiha** [comment]
Plink works on MWE, but issues arise when extracting regions as small as 2 168824768 168850918. The issue is in the plink_slice function, specifically with the command `bed.compute(num_workers=1)` taking too long to run.` bed` is a DataLoader object and I believe the issue has to do specifically with this. A
 similar issue can be found with `.compute(num_workers=1)` [here](https://github.com/pytorch/pytorch/issues/12831).

However, ran notebook directly in JupyterLab and was able to run region 2 168824768 168850918 as well as several other regions, but dealt with an issue with larger regions such as 17 41827392 42969200. The image below shows the error. Attempted both options but extraction failed, possibly because the slices were too large to obtain duplicate copies. This again seems to be an issue with DataLoader objects.
108578744-a2444500-72f1-11eb-8e61-28c6e910fe6a

File: GWAS/Region_Extraction.ipynb [hunk 1]
```diff
@@ -227,14 +228,20 @@
...
-    "        if pb is not None:\n",
-    "            bim = bim.iloc[pb]\n",
-    "            bed = bed[pb,:]\n",
-    "        if pf is not None:\n",
-    "            fam = fam.iloc[pf]\n",
-    "            bed = bed[:,pf]\n",
+    "        print(f'{time.strftime(\"%H:%M:%S\", t)}: slicing geno to get bim')\n",
+    "        \n",
+    "        # slice geno to obtain all geno info within specified region\n",
+    "        bim = p[0].loc[(p[0]['chrom'] == str(region[0])) & (p[0]['pos'] > region[1]) & (p[0]['pos'] < region[2])]\n",
```
Comment (gaow) [inline GWAS/Region_Extraction.ipynb:238]: @tfabiha please use `>=` and `<=` to define region boundaries

**Review by gaow — COMMENTED** [review]
I also dont understand why this chunk of code appeared twice in the notebook:

```
      if geno_file.endswith('.bed'):
            plink = True
            from pandas_plink import read_plink
            geno = read_plink(geno_file)
            print(f'{geno}')
        elif geno_file.endswith('.bgen'):
            plink = False
            from pybgen import PyBGEN
            bgen = PyBGEN(geno_file)
```

one of the occurrence must be redundant.

File: GWAS/Region_Extraction.ipynb (hunk 1 above)
Comment (tfabiha) [reply GWAS/Region_Extraction.ipynb]: Removed the duplication and redefined region boundaries

**Comment by gaow** [comment]
Thanks @tfabiha I'm going to merge this and work on it myself to see if we can resolve the issues you ran into.

**Comment by gaow** [comment]
@tfabiha I have just pushed a commit to implement region extraction for PLINK using `bed_reader` library and not `pandas-plink`. 

@tfabiha @dianacornejo can we try with the current pipeline on `17 41827392 42969200` PLINK file to see if it works?

**Pull request merged**